### PR TITLE
fix a merge forward issue that broke unit association

### PR DESCRIPTION
**Pull request by asmacdo**
This change was not merged forward correctly:

https://github.com/pulp/pulp/commit/b54816cf9f68dc099425b33e5027a82d16e1570d

This also fixes https://pulp.plan.io/issues/1322

### CODE REVIEW

File: server/pulp/server/managers/repo/unit_association_query.py [hunk 1]
```diff
@@ -368,10 +367,7 @@ def _associated_units_by_type_cursor(unit_type_id, criteria, associated_unit_ids
         sort = criteria.unit_sort
 
         if sort is None:
-            try:
-                unit_key = units_controller.get_unit_key_fields_for_type(unit_type_id)
-            except ValueError:
-                unit_key = None
+            sort = [('_id', SORT_ASCENDING)]
```
Comment (seandst) [inline server/pulp/server/managers/repo/unit_association_query.py]: Is there a loss of functionality that comes with hardcoding this here?

Comment (seandst) [inline server/pulp/server/managers/repo/unit_association_query.py]: After poking at it in an IRC discussion: Nope! :D

Comment (asmacdo) [inline server/pulp/server/managers/repo/unit_association_query.py]: This loss of functionality was discussed here: https://github.com/pulp/pulp/pull/1898

The old way was also modified, resulting in the conflict.

**Comment by asmacdo** [comment]
ok test

**Pull request merged**